include/swift/AST/AutoDiff.h

```diff
@@ -32,26 +32,36 @@ enum class DifferentiabilityKind: uint8_t {
 
 class ParsedAutoDiffParameter {
 public:
-  enum class Kind { Named, Self };
+  enum class Kind { Named, Ordered, Self };
 
 private:
   SourceLoc Loc;
   Kind Kind;
   union Value {
-    struct { Identifier Name; }; // Index
+    struct { Identifier Name; }; // Named
+    struct { unsigned int Order; }; // Ordered
```

Comment:
I would suggest naming this property `Index` (and its getter `getIndex()`) because "parameter indices" is a standard term we use in the AD code base. The kind can still be called `Ordered`.

Reply:
What about the static func which returns an instance of `ParsedAutoDiffParameter` I added below. Should it be called `getOrderedParameter` like it is currently or `getIndexParameter`? It has to deal with the enum type so I think keeping it `getOrderedParameter` would be good, but just wanted to verify.

Reply:
`getOrderedParameter()` is good!